src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1021,6 +1026,14 @@ public function setDefinition($id, Definition $definition)
 
         unset($this->aliasDefinitions[$id], $this->removedIds[$id]);
 
+        $bindings = $definition->getBindings();
+        if (!empty($bindings)) {
+            foreach ($bindings as $argument => $binding) {
+                list(, $identifier) = $binding->getValues();
+                $this->bindings[$id][$argument][$identifier] = true;
```

Comment:
Isn't this marking all bindings as used? If I'm correct, no exception will ever be thrown.